File: Signiture-Backend/kotlin/delivery/db/src/main/java/pcy/study/db/storemenu/StoreMenuRepository.java
package pcy.study.db.storemenu;

import org.springframework.data.jpa.repository.JpaRepository;
import pcy.study.db.storemenu.enums.StoreMenuStatus;

import java.util.List;
import java.util.Optional;

public interface StoreMenuRepository extends JpaRepository<StoreMenu, Long> {

    Optional<StoreMenu> findFirstByIdAndStatusOrderByIdDesc(Long id, StoreMenuStatus status);

    List<StoreMenu> findAllByStoreIdAndStatusOrderBySequenceDesc(Long storeId, StoreMenuStatus status);
}

File: Signiture-Backend/kotlin/delivery/db/src/main/kotlin/pcy/study/db/storemenu/StoreMenuRepository.kt
package pcy.study.db.storemenu

import org.springframework.data.jpa.repository.JpaRepository
import pcy.study.db.storemenu.enums.StoreMenuStatus

interface StoreMenuRepository : JpaRepository<StoreMenu, Long> {

    fun findFirstByIdAndStatusOrderByIdDesc(
        id: Long?,
        status: StoreMenuStatus?
    ): StoreMenu?

    fun findAllByStoreIdAndStatusOrderBySequenceDesc(
        storeId: Long?,
        status: StoreMenuStatus?
    ): List<StoreMenu>
}
